### DISCUSSION TITLE
FastDoom 0.8

### DISCUSSION THREAD

**Discussion by viti95** [viti95/FastDoom#26] [General]
- Better RAM usage (also reduced memory footprint)
- More code optimizations
- Compiled with the latest OpenWatcom v2 version
- Stripped episode finale texts from the executable, now are stored in external text files
- Support for Doom II BFG edition. Use the command line parameter "-bfg"
- Smaller executable thanks to UPX compression tool
- New option to render the status bar background a little bit faster. Use the command line "-simplestatusbar". It replaces the status bar background with a simple grey color.
- Two new executables: FDOOMT25, which renders in text mode at 80x25 resolution (CGA, EGA and VGA supported, 16 colors and virtual resolution of 80x50) and FDOOMT50, which renders in text mode at 80x50 resolution (VGA only, 16 colors). Both support triple buffering. These executables have reduced even more the memory footprint as multiple graphics don't need to be loaded (fonts for example). Automap feature isn't supported. This release also includes the WAD ega_pal.wad, which replaces the original colormaps to a better one for text modes.
- New command line parameter "-cga", this let's the new video modes run properly with CGA video cards
- New command line parameter "-fixcolors", this is needed to correct the 256 to 16 colors conversion, as the original Doom palette is too dark for a direct conversion.
- Unified the 386 and 486 executables, the 486 executables were always slower and bigger than the 386 ones. Never figured out why this happens, maybe you can blame OpenWatcom.
- Unified executables between different versions of Doom onto a single one. The supported wads are the following:
   - DOOM.WAD -> DOOM Registered (3 episodes)
   - DOOM1.WAD -> DOOM Shareware (1 episode)
   - DOOMU.WAD -> Ultimate DOOM (4 episodes)
   - DOOM2.WAD -> DOOM II
   - PLUTONIA.WAD -> Final DOOM The Plutonia Experiment
   - TNT.WAD -> Final DOOM TNT Evilution
- New SETUP program! I've found how to edit and compile the original IDSETUP program, and modified it to create a custom version for FastDoom.
- Renamed the configuration file from default.cfg to fdoom.cfg. Also renamed save files from doomsav*.dsg to fdoomsv*.dsg.

<hr /><em>This discussion was created from the release <a href='https://github.com/viti95/FastDoom/releases/tag/0.8'>FastDoom 0.8</a>.</em>

**Comment by flynnsbit** [comment]
Is there something up with the release zip for 0.8?  It seems to have issues with regular decompression on windows, linux unzip can't read it, and 7-zip throws an error but will at least unzip it.  No issues with any other release verison.

**Reply by viti95** [reply]
Ooops! The released zip files where compressed as 7z internally. I've reuploaded them corrected. Thanks for letting me know about the problem!!

**Reply by flynnsbit** [reply]
Thank you! It broke my automation script for updating FastDoom on our FPGA DOS core. :D  Everyone loves FastDoom!
